packages/engine/Source/Core/TerrainEncoding.js

```diff
@@ -24,15 +25,15 @@ const SHIFT_LEFT_12 = Math.pow(2.0, 12.0);
  * @constructor
  *
  * @param {Cartesian3} center The center point of the vertices.
- * @param {AxisAlignedBoundingBox} axisAlignedBoundingBox The bounds of the tile in the east-north-up coordinates at the tiles center.
- * @param {number} minimumHeight The minimum height.
- * @param {number} maximumHeight The maximum height.
- * @param {Matrix4} fromENU The east-north-up to fixed frame matrix at the center of the terrain mesh.
+ * @param {AxisAlignedBoundingBox|undefined} [axisAlignedBoundingBox] The bounds of the tile in the east-north-up coordinates at the tiles center.
+ * @param {number|undefined} [minimumHeight] The minimum height.
```

Comment:
This isn't quite what I meant, sorry if I was unclear. The `[]` alone should indicate that the function argument can be undefined. The problem is the actual class property definitions lower in the file. For example
```js

  /**
   * The minimum height of the tile including the skirts.
   * @type {number}
   */
  this.minimumHeight = minimumHeight;
```
This says that `minimumHeight` is always a number but that's not true if the argument is optional. This is actually a problem with more than just the arguments changed here like `toEnu` and `matrix`. It also could cascade to other places that use these properties (like the `.clone()` function) but it's clearly been working so I'm not sure it's entirely worth the effort to track them all down for this PR.

Reply:
@jjspace thanks for that clarification, I think I fixed it now.
I checked the `.clone` method: the matrices are cloned via `Matrix4.clone`, which cleanly returns `undefined` for an `undefined` argument (although the return type doesn't indicate this). So at least there's not a bug at the moment. Maybe we should revisit the `Matrix?.clone` and `Cartesian?.clone` types at some point.

Reply:
Yea I looked into that too. There's definitely lots of places where this cascades into other functions. Always good to take little steps than no steps, can always clean up more later.